pkg/policies/branch/branch.go

```diff
@@ -356,6 +369,20 @@ func check(ctx context.Context, rep repositories, c *github.Client, owner,
 				pass = false
 			}
 		}
+
+		sp, _, err := rep.GetSignaturesProtectedBranch(ctx, owner, repo, b)
```

Comment:
We want to do:
```
		if err != nil {
			return nil, err
		}
```
Here if we get an error, instead of the "could not determine" below. 

Also, we want to check for expected errors from GitHub, For example they may give a 404 if there is no protection on the branch so a `if rsp != nil && rsp.StatusCode == http.StatusNotFound` check should be done first and set `d.RequireSignedCommits` to `false` and continue on.